fyle_accounting_mappings/views.py

```diff
@@ -79,6 +79,15 @@ def get_queryset(self):
             if source_active and source_active == 'true':
                 params['source__active'] = True
 
+            if source_type == 'CATEGORY':
+                activity_attribute = ExpenseAttribute.objects.filter(
+                    attribute_type='CATEGORY', value='Activity', workspace_id=self.kwargs['workspace_id'], active=True).first()
+                activity_mapping = Mapping.objects.filter(
+                    source_type='CATEGORY', source__value='Activity', workspace_id=self.kwargs['workspace_id']).first()
+                
+                if activity_attribute and not activity_mapping:
+                    params = params & ~Q(source__value='Activity')
+
             mappings = Mapping.objects.filter(**params)
```

Comment:
can we just do `mappings = Mapping.objects.filter(**params, ~Q(source__value='Activity'))`?

Reply:
this way is better no
also works fine

Reply:
this causes unnecessary 2 queries which is checking expense attributes and mappings, we can directly do <> activity